main/views.py

```diff
@@ -1,104 +1,136 @@
 """All views are here."""
 from django.utils import timezone
-from django.shortcuts import render, get_object_or_404, redirect
-from .models import Post, Comment
-from .forms import PostForm, CommentForm
 from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+from django.shortcuts import get_object_or_404, redirect
+from django.core.urlresolvers import reverse_lazy
+from django.views.generic import DetailView, ListView, CreateView, UpdateView, View, DeleteView, TemplateView
+
+from .forms import CommentForm, PostForm
+from .models import Post, Comment
+
 
+class Protected(View):
+    """Protect views that need to show only for authorised users."""
 
-def post_list(request):
+    @method_decorator(login_required)
```

Comment:
We can use '@method_decorator' and there put '@login_required'. But IMHO this not very clear way